checker/AnDO_engine.py

```diff
@@ -18,77 +19,93 @@
         ['Landing', 'sub-enyo', '180116_001_m_enyo_land-001']
     ]
-    to 
+    to
     [
         ['Landing', 'sub-anye', '180116_001_m_anye_land-001', 'source'],
         ['Landing', 'sub-enya', '180116_001_m_enya_land-001', 'source'],
     ]
     Checking for the longest chain with the same sub chain
     """
-    
-    main_list = sorted(nested_list_of_dir, key= lambda sublist: len(sublist)) 
- 
-    # TODO : optimize 
-    i=0
-    j=1
-    while i < len(main_list) -1:
-        if j <= len(main_list) -1:
-            if len(main_list[i]) <= len(main_list[j]):
-                all_in = True
-                for elmt in main_list[i]:
-                    if elmt not in main_list[j]:
-                        all_in = False
-
-                if all_in:
-                    for elmt_to_add in main_list[j]:
-                        if elmt_to_add not in main_list[i]:
-                            main_list[i].append(elmt_to_add)
-                    main_list.pop(j)
-
-                else:
-                    if j < len(main_list) - 1:
-                        j+=1
-                    else:
-                        i +=1
-                        j =i+1
-            else:
-                i+=1
-                j=i+1
-        else:
-            break
+
+    def _test_is_included(my_list_of_lists, list_elem):
+        for my_list_elem in my_list_of_lists:
+            if all([val[0]==val[1] for val in zip(my_list_elem, list_elem)]):
+                return True
+
+        return False
+
+    new_list_of_lists = []
+
+    for list_elem in sorted(nested_list_of_dir,
+                            key= lambda sublist: len(sublist),
+                            reverse=True):
+
+        if not _test_is_included(new_list_of_lists, list_elem):
+            new_list_of_lists.append(list_elem)
+
+    return new_list_of_lists
+
+    #main_list = sorted(nested_list_of_dir, key= lambda sublist: len(sublist))
+
+    ## TODO : optimize
+    #i=0
+    #j=1
+    #while i < len(main_list) -1:
+        #if j <= len(main_list) -1:
+            #if len(main_list[i]) <= len(main_list[j]):
+                #all_in = True
+                #for elmt in main_list[i]:
+                    #if elmt not in main_list[j]:
+                        #all_in = False
+
+                #if all_in:
+                    #for elmt_to_add in main_list[j]:
+                        #if elmt_to_add not in main_list[i]:
+                            #main_list[i].append(elmt_to_add)
+                    #main_list.pop(j)
+
+                #else:
+                    #if j < len(main_list) - 1:
+                        #j+=1
+                    #else:
+                        #i +=1
+                        #j =i+1
+            #else:
+                #i+=1
+                #j=i+1
+        #else:
+            #break
     return main_list
 
 def create_path(directory):
     """
-    New function to get the path given in arg 
+    Function to get the path given in arg
```

Comment:
the name of the function is very generic and not very explicit in saying what it is doing specifically; idem for the description / docstring (which looks also incomplete, no?): "create_path" looks like a function that could be in os.path!

can you change it to something more specific?

Reply:
this does not help understanding what the function does... what are the inputs? what are the outputs? what is the transformation operated by the function? please be more precise!

Reply:
Old version updated in  https://github.com/INT-NIT/AnDOChecker/pull/17/commits/e34ebff5799f2de6b3a0decbe628cc86f9e1164a

Reply:
hmmmm... there must still be something wrong with how we're doing this, coz I see the link to your commit, but not directly its content and the changes you made... so it's impossible to review like this efficiently...
@davidmeunier79 any suggestion?

Reply:
ahhh, I think it's not too bad, I just have to click on the commit number referenced by @Slowblitz ...
(but in my other PRs in nilearn, I don't have to do this I think...)

Reply:
Ye, maybe I did something wrong when I created this PR .